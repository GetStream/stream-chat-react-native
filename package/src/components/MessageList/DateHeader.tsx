import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { useTheme } from '../../contexts/themeContext/ThemeContext';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    alignSelf: 'center',
    borderRadius: 8,
    height: 16,
    justifyContent: 'center',
    marginTop: 8,
    paddingHorizontal: 8,
  },
  text: {
    fontSize: 12,
    textAlign: 'center',
    textAlignVertical: 'center',
  },
});

export type DateHeaderProps = {
  dateString: string;
};

export const DateHeader: React.FC<DateHeaderProps> = ({ dateString }) => {
  const {
    theme: {
      colors: { overlay, white },
      dateHeader: { container, text },
    },
  } = useTheme();

  return (
    <View style={[styles.container, { backgroundColor: overlay }, container]}>
      <Text style={[styles.text, { color: white }, text]}>{dateString}</Text>
    </View>
  );
};
