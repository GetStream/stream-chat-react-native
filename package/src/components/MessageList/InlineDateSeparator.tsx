import React, { useMemo } from 'react';
import { StyleSheet, Text, View } from 'react-native';

import { useTheme } from '../../contexts/themeContext/ThemeContext';
import { useTranslationContext } from '../../contexts/translationContext/TranslationContext';
import { getDateString } from '../../utils/i18n/getDateString';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    alignSelf: 'center',
    borderRadius: 8,
    height: 16,
    justifyContent: 'center',
    marginVertical: 4,
    paddingHorizontal: 8,
  },
  text: {
    fontSize: 12,
    textAlign: 'center',
    textAlignVertical: 'center',
  },
});

/**
 * Props for the `InlineDateSeparator` component.
 */
export type InlineDateSeparatorProps = {
  /**
   * Date to be displayed.
   */
  date?: Date;
};

export const InlineDateSeparator = ({ date }: InlineDateSeparatorProps) => {
  const {
    theme: {
      colors: { overlay, white },
      inlineDateSeparator: { container, text },
    },
  } = useTheme();
  const { t, tDateTimeParser } = useTranslationContext();

  const dateString = useMemo(
    () =>
      getDateString({
        date,
        t,
        tDateTimeParser,
        timestampTranslationKey: 'timestamp/InlineDateSeparator',
      }),
    [date, t, tDateTimeParser],
  );

  return (
    <View
      style={[styles.container, { backgroundColor: overlay }, container]}
      testID='date-separator'
    >
      <Text style={[styles.text, { color: white }, text]}>{dateString}</Text>
    </View>
  );
};
