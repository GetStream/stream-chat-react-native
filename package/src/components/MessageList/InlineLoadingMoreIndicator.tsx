import React from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';

import { usePaginatedMessageListContext } from '../../contexts/paginatedMessageListContext/PaginatedMessageListContext';
import { useTheme } from '../../contexts/themeContext/ThemeContext';

import type { DefaultStreamChatGenerics } from '../../types/types';

const styles = StyleSheet.create({
  activityIndicatorContainer: {
    padding: 10,
    width: '100%',
  },
});

export type InlineLoadingMoreIndicatorPropsWithContext = {
  loadingMore?: boolean;
};

export const InlineLoadingMoreIndicatorWithContext = ({
  loadingMore,
}: InlineLoadingMoreIndicatorPropsWithContext) => {
  const { theme } = useTheme();

  const {
    colors: { accent_blue },
  } = theme;

  if (!loadingMore) {
    return null;
  }

  return (
    <View style={styles.activityIndicatorContainer}>
      <ActivityIndicator color={accent_blue} size='small' />
    </View>
  );
};

const areEqual = (
  prevProps: InlineLoadingMoreIndicatorPropsWithContext,
  nextProps: InlineLoadingMoreIndicatorPropsWithContext,
) => {
  const { loadingMore: prevLoadingMore } = prevProps;
  const { loadingMore: nextLoadingMore } = nextProps;

  const loadingMoreEqual = prevLoadingMore === nextLoadingMore;
  if (!loadingMoreEqual) return false;

  return true;
};

const MemoizedInlineLoadingMoreIndicator = React.memo(
  InlineLoadingMoreIndicatorWithContext,
  areEqual,
) as typeof InlineLoadingMoreIndicatorWithContext;

export const InlineLoadingMoreIndicator = <
  StreamChatGenerics extends DefaultStreamChatGenerics = DefaultStreamChatGenerics,
>() => {
  const { loadingMore } = usePaginatedMessageListContext<StreamChatGenerics>();

  return <MemoizedInlineLoadingMoreIndicator loadingMore={loadingMore} />;
};
