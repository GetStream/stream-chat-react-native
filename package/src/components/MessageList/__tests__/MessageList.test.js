import React from 'react';

import { act, cleanup, render, waitFor } from '@testing-library/react-native';

import { OverlayProvider } from '../../../contexts/overlayContext/OverlayProvider';
import { getOrCreateChannelApi } from '../../../mock-builders/api/getOrCreateChannel';

import { useMockedApis } from '../../../mock-builders/api/useMockedApis';
import dispatchMessageNewEvent from '../../../mock-builders/event/messageNew';
import dispatchTypingEvent from '../../../mock-builders/event/typing';
import { generateChannelResponse } from '../../../mock-builders/generator/channel';
import { generateMember } from '../../../mock-builders/generator/member';
import { generateMessage } from '../../../mock-builders/generator/message';
import { generateUser } from '../../../mock-builders/generator/user';
import { getTestClientWithUser } from '../../../mock-builders/mock';
import { registerNativeHandlers } from '../../../native';
import { Channel } from '../../Channel/Channel';
import { Chat } from '../../Chat/Chat';
import { MessageList } from '../MessageList';

describe('MessageList', () => {
  afterEach(cleanup);

  it('should add new message at bottom of the list', async () => {
    const user1 = generateUser();
    const user2 = generateUser();
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 }), generateMember({ user: user2 })],
      messages: [generateMessage({ user: user1 }), generateMessage({ user: user1 })],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { getByText, queryAllByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel}>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    const newMessage = generateMessage({ user: user2 });
    act(() => dispatchMessageNewEvent(chatClient, newMessage, mockedChannel.channel));

    await waitFor(() => {
      expect(queryAllByTestId('message-notification')).toHaveLength(0);
      expect(getByText(newMessage.text)).toBeTruthy();
    });
  }, 10000);

  it('should render a system message in the list', async () => {
    const user1 = generateUser();
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 })],
      messages: [
        generateMessage({ user: user1 }),
        generateMessage({ type: 'system', user: undefined }),
        generateMessage({ user: user1 }),
      ],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { getByTestId, queryAllByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel}>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    // debug()
    await waitFor(() => {
      expect(queryAllByTestId('error-notification')).toHaveLength(0);
      expect(getByTestId('message-system')).toBeTruthy();
    });
  });

  it('should render deleted message in the list when `deleteMessagesVisibilityType` is set to default(always)', async () => {
    const user1 = generateUser();
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 })],
      messages: [
        generateMessage({ type: 'deleted', user: user1 }),
        generateMessage({ type: 'system', user: undefined }),
        generateMessage({ user: user1 }),
      ],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { getByTestId, queryAllByTestId, queryByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel}>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    await waitFor(() => {
      expect(queryAllByTestId('error-notification')).toHaveLength(0);
      expect(getByTestId('message-deleted')).toBeTruthy();
      expect(queryByTestId('only-visible-to-you')).toBeNull();
    });
  });

  it('should render deleted message in the list when `deleteMessagesVisibilityType` is set to sender', async () => {
    const user1 = generateUser();
    const user2 = generateUser({ id: 'testID' });
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 })],
      messages: [
        generateMessage({ type: 'deleted', user: user2 }),
        generateMessage({ type: 'system', user: undefined }),
        generateMessage({ user: user1 }),
      ],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { getByTestId, queryAllByTestId, queryByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel} deletedMessagesVisibilityType='sender'>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    await waitFor(() => {
      expect(queryAllByTestId('error-notification')).toHaveLength(0);
      expect(queryByTestId('message-deleted')).toBeTruthy();
      expect(getByTestId('only-visible-to-you')).toBeTruthy();
    });
  });

  it('should render deleted message in the list when `deleteMessagesVisibilityType` is set to receiver', async () => {
    const user1 = generateUser();
    const user2 = generateUser({ id: 'testID' });
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 })],
      messages: [
        generateMessage({ user: user2 }),
        generateMessage({ type: 'system', user: undefined }),
        generateMessage({ type: 'deleted', user: user1 }),
      ],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { getByTestId, queryAllByTestId, queryByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel} deletedMessagesVisibilityType='receiver'>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    await waitFor(() => {
      expect(queryAllByTestId('error-notification')).toHaveLength(0);
      expect(getByTestId('message-deleted')).toBeTruthy();
      expect(queryByTestId('only-visible-to-you')).toBeNull();
    });
  });

  it('should render deleted message in the list when `deleteMessagesVisibilityType` is set to never', async () => {
    const user1 = generateUser();
    const user2 = generateUser({ id: 'testID' });
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 })],
      messages: [
        generateMessage({ user: user2 }),
        generateMessage({ type: 'system', user: undefined }),
        generateMessage({ type: 'deleted', user: user1 }),
      ],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { queryAllByTestId, queryByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel} deletedMessagesVisibilityType='never'>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    await waitFor(() => {
      expect(queryAllByTestId('error-notification')).toHaveLength(0);
      expect(queryByTestId('message-deleted')).toBeNull();
      expect(queryByTestId('only-visible-to-you')).toBeNull();
    });
  });

  it('should render deleted message in the list', async () => {
    const user1 = generateUser();
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 })],
      messages: [
        generateMessage({ type: 'deleted', user: user1 }),
        generateMessage({ type: 'system', user: undefined }),
        generateMessage({ user: user1 }),
      ],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { getByTestId, queryAllByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel}>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    await waitFor(() => {
      expect(queryAllByTestId('error-notification')).toHaveLength(0);
      expect(getByTestId('message-deleted')).toBeTruthy();
    });
  });

  it('should render the typing indicator when typing object is non empty', async () => {
    const user1 = generateUser();
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 })],
      messages: [generateMessage({ user: user1 })],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { getByTestId, queryAllByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel}>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    act(() => {
      dispatchTypingEvent(chatClient, user1, mockedChannel.channel);
    });

    await waitFor(() => {
      expect(queryAllByTestId('message-system')).toHaveLength(0);
      expect(queryAllByTestId('error-notification')).toHaveLength(0);
      expect(getByTestId('typing-indicator')).toBeTruthy();
    });
  });

  it('should render the EmptyStateIndicator when no messages loaded', async () => {
    const user1 = generateUser();
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 })],
      messages: [],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { getByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel}>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    await waitFor(() => {
      expect(getByTestId('empty-state')).toBeTruthy();
    });
  });

  it('should render the is offline error', async () => {
    registerNativeHandlers({
      NetInfo: {
        addEventListener: () => () => null,
        fetch: () =>
          new Promise((resolve) => {
            resolve(false);
          }),
      },
    });

    const user1 = generateUser();
    const mockedChannel = generateChannelResponse({
      members: [generateMember({ user: user1 })],
      messages: [generateMessage({ user: user1 })],
    });

    const chatClient = await getTestClientWithUser({ id: 'testID' });
    useMockedApis(chatClient, [getOrCreateChannelApi(mockedChannel)]);
    const channel = chatClient.channel('messaging', mockedChannel.id);
    await channel.watch();

    const { getByTestId, getByText, queryAllByTestId } = render(
      <OverlayProvider>
        <Chat client={chatClient}>
          <Channel channel={channel}>
            <MessageList />
          </Channel>
        </Chat>
      </OverlayProvider>,
    );

    await waitFor(() => {
      expect(queryAllByTestId('message-system')).toHaveLength(0);
      expect(queryAllByTestId('typing-indicator')).toHaveLength(0);
      expect(getByTestId('error-notification')).toBeTruthy();
      expect(getByText('Reconnecting...')).toBeTruthy();
    });

    registerNativeHandlers({
      NetInfo: {
        addEventListener: () => () => null,
        fetch: () =>
          new Promise((resolve) => {
            resolve(true);
          }),
      },
    });
  });
});
