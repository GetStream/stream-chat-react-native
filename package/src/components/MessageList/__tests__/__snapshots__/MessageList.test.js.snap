// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList should render the message list and match snapshot 1`] = `
<View
  keyboardVerticalOffset={86.5}
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <View
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "width": "100%",
          },
          Object {
            "backgroundColor": "#FCFCFC",
          },
          Object {},
        ]
      }
    >
      <View
        testID="message-flat-list"
      >
        <View>
          <View
            animatedStyle={
              Object {
                "value": Object {},
              }
            }
            collapsable={false}
            forwardedRef={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            testID="message-wrapper"
          >
            <View
              animatedStyle={
                Object {
                  "value": Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  },
                }
              }
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "backgroundColor": undefined,
                  "paddingHorizontal": 8,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                animatedStyle={
                  Object {
                    "value": Object {
                      "opacity": 0,
                    },
                  }
                }
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "#FBF4DD",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-end",
                      "flexDirection": "row",
                    },
                    Object {
                      "justifyContent": "flex-start",
                      "marginBottom": 30,
                      "marginTop": 0,
                    },
                    Object {},
                  ]
                }
                testID="message-simple-wrapper"
              >
                <View
                  style={
                    Array [
                      Object {
                        "marginRight": 8,
                      },
                      Object {},
                    ]
                  }
                  testID="message-avatar"
                >
                  <View>
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 16,
                            "height": 32,
                            "width": 32,
                          },
                          Object {},
                          undefined,
                        ]
                      }
                    >
                      <Image
                        accessibilityLabel="avatar-image"
                        onError={[Function]}
                        source={
                          Object {
                            "uri": "https://i.imgur.com/iNaC3K7.jpg",
                          }
                        }
                        style={
                          Array [
                            Object {
                              "borderRadius": 16,
                              "height": 32,
                              "width": 32,
                            },
                            Object {
                              "borderRadius": 16,
                              "height": 32,
                              "width": 32,
                            },
                            undefined,
                          ]
                        }
                        testID="avatar-image"
                      />
                    </View>
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "opacity": 1,
                      "paddingTop": 2,
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderTopLeftRadius": 16,
                            "borderTopRightRadius": 16,
                            "borderWidth": 1,
                            "overflow": "hidden",
                          },
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomLeftRadius": 0,
                            "borderBottomRightRadius": 16,
                            "borderColor": "#ECEBEB",
                          },
                          Object {},
                          Object {},
                        ]
                      }
                      testID="message-content-wrapper"
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "maxWidth": 250,
                              "paddingHorizontal": 16,
                            },
                            Object {},
                            undefined,
                          ]
                        }
                        testID="message-text-container"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignSelf": "stretch",
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignItems": "flex-start",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "justifyContent": "flex-start",
                                "marginBottom": 8,
                                "marginTop": 8,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#000000",
                                }
                              }
                            >
                              Message3
                            </Text>
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "marginTop": 4,
                      }
                    }
                    testID="message-status-time"
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#7A7A7A",
                            "textAlign": "left",
                          },
                          Object {
                            "fontSize": 12,
                          },
                        ]
                      }
                    >
                      8:20 PM
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "borderRadius": 8,
                  "height": 16,
                  "justifyContent": "center",
                  "marginVertical": 4,
                  "paddingHorizontal": 8,
                },
                Object {
                  "backgroundColor": "#00000099",
                },
                Object {},
              ]
            }
            testID="date-separator"
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 12,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {},
                ]
              }
            >
              May 6, 2020
            </Text>
          </View>
        </View>
        <View>
          <View
            animatedStyle={
              Object {
                "value": Object {},
              }
            }
            collapsable={false}
            forwardedRef={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            testID="message-wrapper"
          >
            <View
              animatedStyle={
                Object {
                  "value": Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  },
                }
              }
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "backgroundColor": undefined,
                  "paddingHorizontal": 8,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                animatedStyle={
                  Object {
                    "value": Object {
                      "opacity": 0,
                    },
                  }
                }
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "#FBF4DD",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-end",
                      "flexDirection": "row",
                    },
                    Object {
                      "justifyContent": "flex-start",
                      "marginBottom": 8,
                      "marginTop": 0,
                    },
                    Object {},
                  ]
                }
                testID="message-simple-wrapper"
              >
                <View
                  style={
                    Array [
                      Object {
                        "marginRight": 8,
                      },
                      Object {},
                    ]
                  }
                  testID="message-avatar"
                >
                  <View>
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 16,
                            "height": 32,
                            "width": 32,
                          },
                          Object {},
                          undefined,
                        ]
                      }
                    >
                      <Image
                        accessibilityLabel="avatar-image"
                        onError={[Function]}
                        source={
                          Object {
                            "uri": "https://i.imgur.com/iNaC3K7.jpg",
                          }
                        }
                        style={
                          Array [
                            Object {
                              "borderRadius": 16,
                              "height": 32,
                              "width": 32,
                            },
                            Object {
                              "borderRadius": 16,
                              "height": 32,
                              "width": 32,
                            },
                            undefined,
                          ]
                        }
                        testID="avatar-image"
                      />
                    </View>
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "opacity": 1,
                      "paddingTop": 2,
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderTopLeftRadius": 16,
                            "borderTopRightRadius": 16,
                            "borderWidth": 1,
                            "overflow": "hidden",
                          },
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomLeftRadius": 0,
                            "borderBottomRightRadius": 16,
                            "borderColor": "#ECEBEB",
                          },
                          Object {},
                          Object {},
                        ]
                      }
                      testID="message-content-wrapper"
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "maxWidth": 250,
                              "paddingHorizontal": 16,
                            },
                            Object {},
                            undefined,
                          ]
                        }
                        testID="message-text-container"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignSelf": "stretch",
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignItems": "flex-start",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "justifyContent": "flex-start",
                                "marginBottom": 8,
                                "marginTop": 8,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#000000",
                                }
                              }
                            >
                              Message2
                            </Text>
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "marginTop": 4,
                      }
                    }
                    testID="message-status-time"
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#7A7A7A",
                            "textAlign": "left",
                          },
                          Object {
                            "fontSize": 12,
                          },
                        ]
                      }
                    >
                      8:19 PM
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View>
          <View
            animatedStyle={
              Object {
                "value": Object {},
              }
            }
            collapsable={false}
            forwardedRef={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            testID="message-wrapper"
          >
            <View
              animatedStyle={
                Object {
                  "value": Object {
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                  },
                }
              }
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "backgroundColor": undefined,
                  "paddingHorizontal": 8,
                  "transform": Array [
                    Object {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <View
                animatedStyle={
                  Object {
                    "value": Object {
                      "opacity": 0,
                    },
                  }
                }
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "#FBF4DD",
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-end",
                      "flexDirection": "row",
                    },
                    Object {
                      "justifyContent": "flex-start",
                      "marginBottom": 8,
                      "marginTop": 0,
                    },
                    Object {},
                  ]
                }
                testID="message-simple-wrapper"
              >
                <View
                  style={
                    Array [
                      Object {
                        "marginRight": 8,
                      },
                      Object {},
                    ]
                  }
                  testID="message-avatar"
                >
                  <View>
                    <View
                      style={
                        Array [
                          Object {
                            "alignItems": "center",
                            "justifyContent": "center",
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 16,
                            "height": 32,
                            "width": 32,
                          },
                          Object {},
                          undefined,
                        ]
                      }
                    >
                      <Image
                        accessibilityLabel="avatar-image"
                        onError={[Function]}
                        source={
                          Object {
                            "uri": "https://i.imgur.com/SLx06PP.png",
                          }
                        }
                        style={
                          Array [
                            Object {
                              "borderRadius": 16,
                              "height": 32,
                              "width": 32,
                            },
                            Object {
                              "borderRadius": 16,
                              "height": 32,
                              "width": 32,
                            },
                            undefined,
                          ]
                        }
                        testID="avatar-image"
                      />
                    </View>
                  </View>
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "flex-start",
                      "opacity": 1,
                      "paddingTop": 2,
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={Object {}}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "borderTopLeftRadius": 16,
                            "borderTopRightRadius": 16,
                            "borderWidth": 1,
                            "overflow": "hidden",
                          },
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomLeftRadius": 0,
                            "borderBottomRightRadius": 16,
                            "borderColor": "#ECEBEB",
                          },
                          Object {},
                          Object {},
                        ]
                      }
                      testID="message-content-wrapper"
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "maxWidth": 250,
                              "paddingHorizontal": 16,
                            },
                            Object {},
                            undefined,
                          ]
                        }
                        testID="message-text-container"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignSelf": "stretch",
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignItems": "flex-start",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "justifyContent": "flex-start",
                                "marginBottom": 8,
                                "marginTop": 8,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#000000",
                                }
                              }
                            >
                              Message1
                            </Text>
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                        "marginTop": 4,
                      }
                    }
                    testID="message-status-time"
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#7A7A7A",
                            "textAlign": "left",
                          },
                          Object {
                            "fontSize": 12,
                          },
                        ]
                      }
                    >
                      8:18 PM
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "borderRadius": 8,
                  "height": 16,
                  "justifyContent": "center",
                  "marginVertical": 4,
                  "paddingHorizontal": 8,
                },
                Object {
                  "backgroundColor": "#00000099",
                },
                Object {},
              ]
            }
            testID="date-separator"
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 12,
                    "textAlign": "center",
                    "textAlignVertical": "center",
                  },
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {},
                ]
              }
            >
              May 5, 2020
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "top": 0,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "left": 0,
              "paddingVertical": 4,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "#7A7A7AE6",
            },
            Object {},
          ]
        }
        testID="error-notification"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
              },
              Object {},
            ]
          }
        >
          Reconnecting...
        </Text>
      </View>
    </View>
  </View>
</View>
`;
